### PR TITLE
Add constant flag to use the new payment service

### LINKED ISSUE #6675: Payment Process: Use the payment service
### Description

Add a `process_payment` method to the payment service.

To be conditionally called from `WC_Payment_Gateway_WCPay::process_payment()`, avoiding the rest of the original method.

Initial implementation should be based on a locally defined constant (ex. `WCPAY_NEW_PROCESS`), checked within a method of the service.

### Acceptance criteria

- Processing payments returns an empty array for now.
- The service is available as a dependency of the gateway.
- Gateway tests work.
- When `WCPAY_NEW_PROCESS` is not defined, the gateway should work as usual.

### Testing instructions

Tests work.

### Additional context

Epic: https://github.com/Automattic/woocommerce-payments/issues/6673
Project: paJDYF-9hL-p2

### PR DESCRIPTION
**Pull request by htdat**
Fixes #6675

#### Changes proposed in this Pull Request

- Add constant flag to conditionally use the new payment service in `WC_Payment_Gateway_WCPay::process_payment()`.

#### Testing instructions

- Test the legacy card flow (no UPE at all), everything is still the same.
- Add `define( 'WCPAY_NEW_PROCESS', true );` to the main file of plugin `woocommerce-payments.php` or `wp-config.php`, and try to check out. See this error:  
Markup on 2023-08-02 at 12:13:22

- Ideally, test both classic and block checkouts. 

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6873` or branch name `rpp/6675-use-payment-service` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=rpp/6675-use-payment-service) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 0c79151d9c23b44573e714564dfee33d0b97aaae
- Build time: 2023-08-02 10:47:20 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by htdat** [comment]
Ready for review.

File: src/Internal/Service/PaymentProcessingService.php [hunk 1]
```diff
@@ -7,9 +7,21 @@
 
 namespace WCPay\Internal\Service;
 
+use Exception;
```
Comment (RadoslavGeorgiev) [inline src/Internal/Service/PaymentProcessingService.php]: Note to self: We might want to introduce a different exception class for the payment process. It's alright for now.

**Review by RadoslavGeorgiev — APPROVED** [review]
LGTM :shipit: 

Once merged, I'll update https://github.com/Automattic/woocommerce-payments/pull/6862 to load the service through `wcpay_get_container()`.

File: src/Internal/Service/PaymentProcessingService.php (hunk 1 above)
Comment (htdat) [reply src/Internal/Service/PaymentProcessingService.php]: Definitely! Good point, added a comment besides it 42a3a4ceea33b619bd9a5cb844e532b0208bf41b

**Pull request merged**